### PR TITLE
Fix null pathways when a naked throw is issued at the PSIC console

### PR DESCRIPTION
**Pull request by JustinGrote**
Fixes https://github.com/PowerShell/vscode-powershell/issues/3677

### CODE REVIEW

**Comment by JustinGrote** [comment]
Works as expected now, the color styling is another issue
image

File: src/PowerShellEditorServices/Utility/PSCommandExtensions.cs [hunk 1]
```diff
@@ -55,10 +55,13 @@ public static PSCommand AddDebugOutputCommand(this PSCommand psCommand)
...
-            Command lastCommand = psCommand.Commands[psCommand.Commands.Count - 1];
-            lastCommand.MergeMyResults(PipelineResultTypes.Error, PipelineResultTypes.Output);
-            lastCommand.MergeMyResults(PipelineResultTypes.Information, PipelineResultTypes.Output);
+            if (psCommand.Commands.Count > 0)
+            {
+                // We need to do merge errors and output before rendering with an Out- cmdlet
+                Command lastCommand = psCommand.Commands[psCommand.Commands.Count - 1];
+                lastCommand.MergeMyResults(PipelineResultTypes.Error, PipelineResultTypes.Output);
+                lastCommand.MergeMyResults(PipelineResultTypes.Information, PipelineResultTypes.Output);
+            }
```
Comment (SeeminglyScience) [inline src/PowerShellEditorServices/Utility/PSCommandExtensions.cs:64]: I'm kinda curious what circumstance we're calling this method without any commands added though 🤔

Comment (JustinGrote) [reply src/PowerShellEditorServices/Utility/PSCommandExtensions.cs:64]: With `throw`, there was no "previous command output" and that's where the null was coming from.

Comment (JustinGrote) [reply src/PowerShellEditorServices/Utility/PSCommandExtensions.cs:64]: https://github.com/JustinGrote/PowerShellEditorServices/blob/ceb20e8f47754f5c829c6f4ae9c102241f7e59da/src/PowerShellEditorServices/Services/PowerShell/Execution/SynchronousPowerShellTask.cs#L106-L109

**Comment by andschwa** [comment]
@JustinGrote Did you mean to close this?

**Comment by JustinGrote** [comment]
@andschwa I probably just deleted the branch during a cleanup accidentally, should still be valid.

**Review by andschwa — APPROVED** [review]
Heh, I thought there would be more too this for some reason. Great and simple fix!

**Comment by andschwa** [comment]
@JustinGrote I'm going to see if I can get the old PowerShell context service tests online and add a regression test for this.

**Comment by andschwa** [comment]
Ok, I enabled those tests! After they're merged I'll try to write a regression test for this (or you are free to take a stab at it) and then merge it.

**Comment by andschwa** [comment]
Closing since I got this change in!

**Pull request closed without merging**